### ISSUE TITLE
Error while solving linear system with Diagonal sparse complex rational matrices

### ISSUE THREAD

**Issue by eschnett** [JuliaSparse/SparseArrays.jl#223]
This fails:
```Julia
using LinearAlgebra
using SparseArrays
A = Diagonal(sparsevec([1//1+0im]))
A / A
```
The error message is `ERROR: DivideError: integer division error`. This could be related to https://github.com/JuliaSparse/SparseArrays.jl/issues/221.

Note that the very similar `A \ A` works fine.

**Comment by eschnett** [comment]
I am using
```
julia> versioninfo()
Julia Version 1.9.0-DEV.1125
Commit e1fa6a51e4* (2022-08-13 01:32 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin21.6.0)
  CPU: 16 × Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-14.0.5 (ORCJIT, skylake)
  Threads: 8 on 16 virtual cores
```

**Comment by dkarrasch** [comment]
This works now. I believe it was fixed by #222.
